### DISCUSSION TITLE
Asset not found

### DISCUSSION THREAD

**Discussion by tobithe1st** [AlmasB/FXGL#1262] [Q&A] [answered]
Hey,

Im just getting started with FXGL and already have problems getting the TileMap set up.

```
    @Override
    protected void initGame(){
        FXGL.getGameWorld().addEntityFactory(new KeplerTownFactory());

        FXGL.setLevelFromMap("test.tmx");
    }

    public static void main(String[] args) {
        launch(args);
    }
}

```
After starting the application I get the following exception:

![image](https://user-images.githubusercontent.com/118936956/229801227-0a5d7271-fd84-4ec7-811a-f5bda8fc7f2d.png)

I already tried all possible directories - nothing. When choosing a json file, the AssetLoader finds it ind resources/assets/json , but is also throwing an exception:

![image](https://user-images.githubusercontent.com/118936956/229801078-3ac4b5ee-5aa8-4c9f-8811-a024b46a8a03.png)

Map was created with Tiled.

**Comment by AlmasB** [comment]
Hi, the above code for loading `.tmx` seems fine. Please ensure the file `"test.tmx"` is located under `/assets/levels/`

**Reply by tobithe1st** [reply]
It's here:
![image](https://user-images.githubusercontent.com/118936956/229895401-5c897fe4-f2ad-4513-a513-4ca43b84b2c3.png)
![image](https://user-images.githubusercontent.com/118936956/229895552-6bb6f432-dfe4-47e4-a650-d44b1e31f676.png)

**Reply by AlmasB** [reply]
Everything looks fine, please can you try running your project from the command line and/or using a different IDE. If it runs, then the IDE is not copying the resources directory to the target (where compiled code is) directory. If it fails, you can try uploading a minimal example to a GitHub repo for further checks.

**Reply by tobithe1st** [reply]
https://github.com/tobithe1st/KeplerTown
Thx for your help, really appreciate it!

**Reply by AlmasB** [reply, marked as answer]
1. The line in module-info.java: `module com.keplertown.keplertown {` needs to be `open module ...` to allow FXGL to have access to resources.

2. `.tmx` file format is not fully supported. Of particular note with this project are relative paths and infinite maps, so I re-generated the map to workaround these, see attached as `test.txt` (you will need to rename back to .tmx)
[test.txt](https://github.com/AlmasB/FXGL/files/11162785/test.txt)

3. All tileset images will need to be located in `assets/levels/`, where `test.tmx` lives.

I've opened a new feature request for the unsupported features.

The result is something like this:

![Screenshot 2023-04-05 21 19 41](https://user-images.githubusercontent.com/3594742/230202414-13138cca-02cb-4f1b-ae09-28e6c281725b.png)

**Reply by tobithe1st** [reply]
thanks for your help!
